src/odemis/gui/cont/tabs.py

```diff
@@ -4389,7 +4393,481 @@ def _on_align_move_done(self, f=None):
+        for viewport in self.panel.pnl_two_streams_grid.viewports:
+            if viewport.view.stream_classes is acqstream.FIBStream:
+                for FIB_line_overlay in viewport.canvas.view_overlays:
+                    if isinstance(FIB_line_overlay, CenteredLineOverlay):
+                        FIB_line_overlay.shape = CROSSHAIR
+                        break
+
+        doc_path = pkg_resources.resource_filename("odemis.gui", "doc/enzel_sem_alignment.html")
+        self.panel.html_alignment_doc.LoadPage(doc_path)
+
+        self.panel.controls_step_size_slider.SetRange(1e-6,
+                                                      min(50e-6, abs(self._sem_stream.emitter.beamShift.range[1][0])))
+        self.panel.controls_step_size_slider.set_position_value(10e-6)
+
+        self.panel.pnl_z_align_controls.Show(False)
+        self.panel.pnl_sem_align_controls.Show(True)
+        self.panel.pnl_flm_align_controls.Show(False)
+        fix_static_text_clipping(self.panel)
+
+    def _sem_alignment_controls(self, axis, proportion, evt):
+        """
+        Links the SEM alignments mode control buttons to the stage.
+
+        :param axis (str): axis which is controlled by the button.
+        :param proportion (int): direction of the movement represented by the button (-1/1)
+        :param evt (GenButtonEvent): Clicked event
+        """
+        evt.theButton.Enabled = False
+        # Beam shift control
+        self._sem_move_beam_shift_rel({axis: proportion * self.tab_data_model.step_size.value})
+        self.stream_bar_controller.refreshStreams((self._sem_stream,))
+        evt.theButton.Enabled = True
+
+    def _sem_move_beam_shift_rel(self, shift):
+        """
+        Provides relative control of the beam shift similar to the MoveRel functionality of the stages.
+
+        :param shift (dict --> float): Relative movement of the beam shift with the axis as keys (x/y)
+        """
+        beamShiftVA = self._sem_stream.emitter.beamShift
+        try:
+            if "x" in shift:
+                beamShiftVA.value = (beamShiftVA.value[0] + shift["x"], beamShiftVA.value[1])
+        except IndexError:
+            show_message(wx.GetApp().main_frame,
+                         "Reached the limits of the beam shift, cannot move any further in x direction.",
+                         timeout=3.0, level=logging.WARNING)
+            logging.error("Reached the limits of the beam shift, cannot move any further in x direction.")
+
+        try:
+            if "y" in shift:
+                beamShiftVA.value = (beamShiftVA.value[0], beamShiftVA.value[1] + shift["y"])
+        except IndexError:
+            show_message(wx.GetApp().main_frame,
+                         "Reached the limits of the beam shift, cannot move any further in y direction.",
+                         timeout=3.0, level=logging.WARNING)
+            logging.error("Reached the limits of the beam shift, cannot move any further in y direction.")
```

Comment:
Same code block as above except in x & y, maybe combine them as function?

Reply:
I think it doesn't reduce the number of lines that much plus it would actually reduce the readability of the code.